File: vue-vben-admin/apps/web-naive/src/api/enum/ParamTypeEnum.ts
export enum ParamTypeEnum {
  String = 1,    // 文本
  Number = 2,    // 数值
  Date = 3,    // 日期
  DateTime = 4,    // 日期时间
  Time = 5,    // 时间
  Password = 6,    // 密码
  Option = 7,    // 选项
  Customer = 8,    // 自定义
}

export const ParamTypeEnumOption = [
  { label: '文本', value: 1 },
  { label: '数值', value: 2 },
  { label: '日期', value: 3 },
  { label: '日期时间', value: 4 },
  { label: '时间', value: 5 },
  { label: '密码', value: 6 },
  { label: '选项', value: 7 },
  { label: '自定义', value: 8 },
];

export function getParamTypeLabel(value: number): string | undefined {
  const item = ParamTypeEnumOption.find(x => x.value === value);
  return item?.label;
}

File: src/SimpleApp/Simple.AdminApplication/SysMng/Enums/ParamTypeEnum.cs
using System.ComponentModel;

namespace Simple.AdminApplication.SysMng.Enums
{
    /// <summary>
    /// 参数类型枚举
    /// </summary>
    [GenerateTypeScript]
    public static class ParamTypeEnum
    {
        /// <summary>
        /// 文本
        /// </summary>
        [Description("文本")]
        public static readonly int String = 1;

        /// <summary>
        /// 数值
        /// </summary>
        [Description("数值")]
        public static readonly int Number = 2;

        /// <summary>
        /// 日期
        /// </summary>
        [Description("日期")]
        public static readonly int Date = 3;

        /// <summary>
        /// 日期时间
        /// </summary>
        [Description("日期时间")]
        public static readonly int DateTime = 4;

        /// <summary>
        /// 时间
        /// </summary>
        [Description("时间")]
        public static readonly int Time = 5;

        /// <summary>
        /// 密码
        /// </summary>
        [Description("密码")]
        public static readonly int Password = 6;

        /// <summary>
        /// 选项
        /// </summary>
        [Description("选项")]
        public static readonly int Option = 7;

        /// <summary>
        /// 自定义
        /// </summary>
        [Description("自定义")]
        public static readonly int Customer = 8;
    }
}